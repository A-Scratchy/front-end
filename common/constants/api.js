import { get, post, patch } from 'common/utils/api-utils';
import { coerceEmptyStringToUndefined } from 'common/utils/string-utils';

/* GET REQUESTS */
export const getCodeSchoolsPromise = () => get('code_schools');
export const getMentorPromise = id => get(`mentors/${id}`);
export const getMentorsPromise = () => get('mentors');
export const getRequestsPromise = () => get('requests');
export const getScholarshipPromise = id => get(`scholarships/${id}`);
export const getScholarshipsPromise = () => get('scholarships');
export const getServicesPromise = () => get('services');
export const getTeamMembersPromise = () => get('team_members');

/* POST REQUESTS */
export const createUser = ({ email, password, firstName, lastName, zipcode }) =>
  post('users', {
    user: {
      email,
      password,
      first_name: firstName,
      last_name: lastName,
      zip: zipcode,
    },
  }).then(({ data }) => data);

export const loginUser = ({ email, password }) =>
  post('sessions', {
    user: {
      email,
      password,
    },
  }).then(({ data }) => data);

export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service }) =>
  post('requests', {
    request: {
      details: additionalDetails,
      language,
      requested_mentor_id: mentor,
      service_id: service,
    },
  });

/* PATCH REQUESTS */
export const updateUser = ({
  branchOfService,
  companyName,
  companyRole,
  disciplines,
  doesWantMentor, // TODO: Use in form
  doesWantScholarshipInfo, // TODO: Use in form
  doesWantToVolunteer, // TODO: Use in form
  employmentStatus,
  militaryStatus,
  payGrade,
  programmingLanguages,
  yearsOfService,
}) => {
  const interestItems = [...programmingLanguages, ...disciplines];

  // Interests could be an array of many filled strings, one filled string, or one empty string
  const interests =
    interestItems.length > 1
      ? interestItems.join(', ')
      : [coerceEmptyStringToUndefined(interestItems[0])];

  return patch('users', {
    user: {
      branch_of_service: coerceEmptyStringToUndefined(branchOfService),
      company_name: coerceEmptyStringToUndefined(companyName),
      company_role: coerceEmptyStringToUndefined(companyRole),
      mentor: doesWantMentor,
      scholarship_info: doesWantScholarshipInfo,
      volunteer: doesWantToVolunteer,
      employment_status: coerceEmptyStringToUndefined(employmentStatus),
      military_status:
        militaryStatus === 'civilian' ? undefined : coerceEmptyStringToUndefined(militaryStatus),
      pay_grade: coerceEmptyStringToUndefined(payGrade),
      interests,
      years_of_service: coerceEmptyStringToUndefined(yearsOfService),
    },
  }).then(({ data }) => data);
};

export const patchUpdateMentorRequestPromise = ({ request, status, mentor }) =>
  patch(`requests/${request}`, {
    request: {
      status,
      mentor,
    },
  });
