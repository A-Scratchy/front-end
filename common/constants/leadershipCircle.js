// Instructions:

// Below is a list. To do this, you must do the following:
// 1:
// 2:
// 3:
// 4:
// 5:
// 6:
// 7:

const honorRoll = [{}];

export default honorRoll;
