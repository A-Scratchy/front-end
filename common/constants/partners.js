const partners = [
  {
    name: 'APEX Systems',
    logo: 'partnerLogos/apex_systems.png',
    url: 'https://apexsystems.com',
  },
  {
    name: 'GitHub',
    logo: 'partnerLogos/github.png',
    url: 'https://github.com',
  },
  {
    name: '1Password',
    logo: 'partnerLogos/1password.png',
    url: 'https://1password.com/',
  },
  {
    name: 'Docker',
    logo: 'partnerLogos/docker.png',
    url: 'https://docker.com',
  },
  {
    name: 'Oracle',
    logo: 'partnerLogos/oracle.png',
    url: 'https://oracle.com',
  },
  {
    name: "O'Reilly Media",
    logo: 'partnerLogos/oreilly.png',
    url: 'https://oreilly.com',
  },
  {
    name: 'HackerRank',
    logo: 'partnerLogos/hackerrank.png',
    url: 'https://hackerrank.com',
  },
  {
    name: 'Zapier',
    logo: 'partnerLogos/zapier.png',
    url: 'https://zapier.com',
  },
];

export default partners;
