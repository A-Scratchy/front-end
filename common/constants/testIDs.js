export const ALERT = 'ALERT';
export const ALERT_CLOSE_BUTTON = 'Close Alert Button';
export const BUTTON = 'BUTTON';
export const CHANGE_PASSWORD_FORM_ERROR = 'CHANGE_PASSWORD_FORM_ERROR';
export const FLAT_CARD_IMAGE = 'FLAT_CARD_IMAGE';
export const INPUT = 'INPUT';
export const INPUT_ERROR = 'INPUT_ERROR';
export const INPUT_FEEDBACK_GROUPING = 'INPUT_FEEDBACK_GROUPING';
export const LABEL = 'LABEL';
export const SCREEN_READER_ONLY = 'SCREEN_READER_ONLY';
