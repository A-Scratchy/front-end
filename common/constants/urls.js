/* eslint-disable max-len */
export const s3 = 'https://s3.amazonaws.com/operationcode-assets/';
export const donateLink = 'https://secure.lglforms.com/form_engine/s/BRtP7QUKyHOyEYsZROsRew';
export const leadershipCircleLink =
  'https://secure.lglforms.com/form_engine/s/L428AQ2rrsFJQyy5Fbglvg';
export const footerLinks = {
  column1: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column2: [
    {
      url: '/who_we_serve',
      title: 'Who We Serve',
    },
    {
      url: '/code_schools',
      title: 'Code Schools',
    },
    {
      url: '/job_board',
      title: 'Job Board',
    },
    {
      url: '/events',
      title: 'Events',
    },
    {
      url: '/blog',
      title: 'Blog',
    },
  ],
  column3: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column4: [
    {
      url: '/resources',
      title: 'Resources',
    },
    {
      url: '/press',
      title: 'Press',
    },
    {
      url: '/branding',
      title: 'Branding',
    },
  ],
  legal: [
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Terms of Use',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Cookies',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: 'https://www.iubenda.com/privacy-policy/8174861',
      title: 'Privacy',
    },
  ],
};
