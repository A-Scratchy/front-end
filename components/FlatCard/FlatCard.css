.FlatCard {
  display: flex;
  flex-direction: column;
}

.imageWrapper {
  display: flex;
  justify-content: center;
}

.image {
  order: 1;
  height: 200px;
  width: 200px;
}

.header {
  order: 2;
  text-align: center;
}

.subHeader {
  order: 3;
  text-align: center;
}

.body {
  order: 4;
}
