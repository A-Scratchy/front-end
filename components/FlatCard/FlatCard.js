import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './FlatCard.css';

FlatCard.propTypes = {
  className: PropTypes.string,
  imageSource: PropTypes.string.isRequired,
  imageAlt: PropTypes.string.isRequired,
  header: PropTypes.node,
  content: PropTypes.node.isRequired,
};

FlatCard.defaultProps = {
  className: undefined,
  header: undefined,
};

function FlatCard({ className, imageSource, imageAlt, header, content }) {
  return (
    <article className={classNames(styles.FlatCard, className)}>
      <div className={styles.borderContainer}>
        {header && <div className={styles.header}>{header}</div>}
        <div className={styles.rowCenter}>
          <img className={styles.image} src={imageSource} alt={imageAlt} />
        </div>
        {header && <hr className={styles.divider} />}
        <div className={styles.content}>{content}</div>
      </div>
    </article>
  );
}

export default FlatCard;
