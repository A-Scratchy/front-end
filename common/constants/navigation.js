import { donateLink } from 'common/constants/urls';

// links shared between nav and footer (no duplicate code)
const contactLink = {
  name: 'Contact Us',
  href: '/contact',
};

const faqLink = {
  name: 'FAQ',
  href: '/faq',
};

const getInvolvedLink = {
  name: 'Get Involved',
  href: '/get_involved',
};

const eventsLink = {
  name: 'Events',
  href: '/events',
};

export const whoWeServeSection = {
  name: 'Who We Serve',
  href: '/who_we_serve',
  shouldPrefetch: false,
  sublinks: [
    {
      name: 'Member Login',
      href: '/login',
    },
    {
      name: 'Join',
      href: '/join',
    },
  ],
};

const logoutLink = {
  name: 'Logout',
  href: '/login?loggedOut=true',
  shouldPrefetch: false,
  sublinks: [],
};

export const navItems = [
  {
    name: 'About Us',
    href: '/about',
    shouldPrefetch: false,
    sublinks: [contactLink, faqLink],
  },
  {
    ...eventsLink,
    shouldPrefetch: false,
    sublinks: [],
  },
  {
    ...getInvolvedLink,
    shouldPrefetch: false,
    sublinks: [
      // {
      //   name: 'Mentoring',
      //   href: '/mentoring',
      // },
      {
        name: 'Sponsorship',
        href: '/sponsorship',
      },
      {
        name: 'Leadership Circle',
        href: '/leadership_circle',
      },
    ],
  },
];

export const loggedInNavItems = [...navItems, logoutLink];
export const loggedOutNavItems = [...navItems.slice(0, 1), whoWeServeSection, ...navItems.slice(1)];

export const footerItems = {
  column1: [
    {
      href: '/about',
      name: 'About Us',
    },
    contactLink,
    faqLink,
  ],
  column2: [
    {
      href: '/code_schools',
      name: 'Code Schools',
    },
    {
      href: '/jobs',
      name: 'Job Board',
    },
    eventsLink,
    {
      href: '/who_we_serve',
      name: 'Who We Serve',
    },
  ],
  column3: [
    getInvolvedLink,
    // {
    //   href: '/mentoring',
    //   name: 'Become A Mentor',
    // },
    {
      href: '/history',
      name: 'History',
    },
    {
      href: donateLink,
      name: 'Donate',
      analyticsEventLabel: 'Donate',
    },
  ],
  column4: [
    // {
    //   href: '/resources',
    //   name: 'Resources',
    // },
    {
      href: '/press',
      name: 'Press',
    },
    {
      href: '/branding',
      name: 'Branding',
    },
    {
      href: '/team',
      name: 'Team',
    },
    // {
    //   href: '/blog',
    //   name: 'Blog',
    // },
  ],
  legal: [
    {
      href: '/terms',
      name: 'Terms of Use',
    },
    // TODO: Create a link for this
    // {
    //   href: 'https://www.anotherfakeurl.com/',
    //   name: 'Cookies',
    //   analyticsEventLabel: 'Cookies',
    // },
    {
      // NOTE: If you change this route, please update the redirect in `now.json` as well
      href: 'https://www.iubenda.com/privacy-policy/8174861',
      name: 'Privacy',
      analyticsEventLabel: 'Privacy',
    },
  ],
};
