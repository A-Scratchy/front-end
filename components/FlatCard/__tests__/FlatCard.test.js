/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import { s3 } from 'common/constants/urls';
import FlatCard from '../FlatCard';

const testImageUrl = `${s3}headshots/david_molina.jpg`;

describe('FlatCard', () => {
  it('should render with required props', () => {
    createSnapshotTest(
      <FlatCard imageSource={testImageUrl} imageAlt="Image caption" headerText="Tester Person">
        Test
      </FlatCard>,
    );
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <FlatCard
        className="test-class"
        imageSource={testImageUrl}
        imageAlt="Image Caption"
        headerText="Tester Person"
        subHeaderText="Leader of the Company"
        bodyText="Description of the person shown on the card."
      >
        Test
      </FlatCard>,
    );
  });
});
