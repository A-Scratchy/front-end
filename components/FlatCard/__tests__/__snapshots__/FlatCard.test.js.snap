// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatCard should render with many props assigned 1`] = `
<article
  className="FlatCard test-class"
>
  <h1
    className="header"
  >
    Tester Person
  </h1>
  <h6
    className="subHeader"
  >
    Leader of the Company
  </h6>
  <div
    className="imageWrapper"
  >
    <img
      alt="Image Caption"
      className="image"
      src="https://s3.amazonaws.com/operationcode-assets/headshots/david_molina.jpg"
    />
  </div>
  <hr />
  <p
    className="body"
  >
    Description of the person shown on the card.
  </p>
</article>
`;

exports[`FlatCard should render with required props 1`] = `
<article
  className="FlatCard"
>
  <h1
    className="header"
  >
    Tester Person
  </h1>
  <div
    className="imageWrapper"
  >
    <img
      alt="Image caption"
      className="image"
      src="https://s3.amazonaws.com/operationcode-assets/headshots/david_molina.jpg"
    />
  </div>
  <hr />
  <p
    className="body"
  />
</article>
`;
